dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/intermediate/components/ComponentBase.kt

```diff
@@ -32,6 +34,21 @@ open class ComponentBase(
      */
     var explanation: String? = null
 
+    /**
+     * The unit of a string component
+     */
+    var unit: String? = null
+
+    /**
+     * The unit of a string component
+     */
+    var uploadComponentName: String? = null
+
+    /**
+     * The unit of a string component
+     */
+    var required: Boolean? = null
```

Comment:
We already have a field "isNullable" which may encode the same information. Check if this is truly required.

Reply:
Fixed. Added proxy field "isRequired" to avoid confusion in the future.